internal/luks2/cryptsetup.go

```diff
@@ -76,6 +107,34 @@ func cryptsetupCmd(stdin io.Reader, callback func(cmd *exec.Cmd) error, args ...
 	return nil
 }
 
+// DetectCryptsetupFeatures returns the features supported by the cryptsetup binary
+// on this system.
+func DetectCryptsetupFeatures() Features {
+	internal.FeaturesOnce.Do(func() {
+		features = 0
+
+		cmd := exec.Command("cryptsetup", "--version")
+		out, err := cmd.CombinedOutput()
+		if err == nil {
+			var major, minor, patch int
+			n, _ := fmt.Sscanf(string(out), "cryptsetup %d.%d.%d", &major, &minor, &patch)
+			if n == 3 {
+				if major >= 3 || (major == 2 && minor >= 1) {
+					features |= FeatureHeaderSizeSetting
+				}
+				if major >= 3 || (major == 2 && minor >= 1) || (major == 2 && minor == 0 && patch >= 3) {
+					features |= FeatureTokenImport
+				}
+			}
+		}
+		if err := cryptsetupCmd(nil, nil, "--test-args", "token", "import", "--token-id", "0",
```

Comment:
is --test-args an undocumented (I don't see it in the man page) but supported option?

Reply:
It is undocumented. It's currently used by cryptsetup's unit tests for validating combinations of commandline arguments.

Reply:
It is documented in the `--help` output though:
```
      --test-args                       Do not run action, just validate all
                                        command line parameters
```